src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -73,6 +74,26 @@ internal sealed class ServiceBusListener : IListener, IScaleMonitorProvider
                 _messageProcessor = _messagingProvider.CreateMessageProcessor(entityPath, _serviceBusAccount.ConnectionString);
             }
             _serviceBusOptions = config;
+
+            _details = new Lazy<string>(() =>
+            {
+                string endpoint = String.Empty;
+                if (_clientEntity != null)
+                {
+                    endpoint = _clientEntity.ServiceBusConnection.Endpoint.ToString();
+                } 
+                else if (_receiver.IsValueCreated)
+                {
+                    endpoint = _receiver.Value.ServiceBusConnection.Endpoint.ToString();
+                } 
+                else if (_sessionClient.IsValueCreated)
+                {
+                    endpoint = _sessionClient.Value.ServiceBusConnection.Endpoint.ToString();
+                }

```

Comment:
Do you mind clarifying why this is necessary here but wasn't in track2?

Reply:
For track1 it's little bit messy as we have 3 different clients for 3 different scenarios, for track2 it's only one client

Reply:
Thanks :)